posix/common

```diff
@@ -12,6 +12,11 @@ if [ ! -d .git ]; then
   git config --global --add safe.directory '*'
   git remote add origin ${DRONE_REMOTE_URL}
   set +x
+else
+  set -x
+  git config --global --add safe.directory '*'
```

Comment:
If your setup relies on specific directories being treated as unsafe by Git, this change might affect other workflows. A better approach would be to limit this configuration to the specific repository by using --local instead of --global.

Reply:
If the remote origin already exists but the URL isn't set correctly, git remote add origin would fail, potentially causing the script to break or behave unexpectedly.

Reply:
Thank you! Replaced `git config --global --add safe.directory '*'` by `git config --local --add safe.directory '*'`, as suggested. I've also wrapped the `git remote set-url origin` logic into another function, which checks if `origin` exists. If the `origin` remote exists, it calls `git remote set-url origin "${origin_url}"`, otherwise it calls `git remote add origin "${origin_url}"`

Reply:
@harshjain12 I've noticed `git config --local --add safe.directory '*'` does not work to solve the dubious ownership error. It seems like git doesn't consider the current folder to be a git repo, if the `.git` file is owned by a different user. Here is an example:
![image](https://github.com/user-attachments/assets/54552924-953d-468e-a2a5-c64b3fece21d)


I've reverted the changes in this PR to use `git config --global --add safe.directory '*'` again. 

I don't think this would affect other workflows. The global git configurations are stored in the container's $HOME directory, which is not shared between containers, so this would not affect any subsequent container steps. Do you see any issues?